src/pool-cl/libraries/SwapMath.sol

```diff
@@ -83,7 +83,7 @@ library SwapMath {
                 // we didn't reach the target, so take the remainder of the maximum input as fee
                 feeAmount = uint256(amountRemaining) - amountIn;
             } else {
-                feeAmount = FullMath.mulDivRoundingUp(amountIn, feePips, 1e6 - feePips);
+                feeAmount = feePips == 1e6 ? amountIn : FullMath.mulDivRoundingUp(amountIn, feePips, 1e6 - feePips);
```

Comment:
just to confirm: can feePips be 1_000_000? seems meaningless to allow it

Reply:
yes, possible

say  protocolFee = whatever, and lpFee = 100%

swapFee = protocolFee + (1 - protocolFee) * 100% = 1